### PR TITLE
Add admin command tests for setClusterParameter

### PR DESCRIPTION
**Pull request by vic-tsang**
This PR contains: 
- admin command tests for setClusterParameter

Ref:
- Issue #428

### CODE REVIEW

File: documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_core_behavior.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        _set_param(collection, ALT_VALUE_2)
+        result = execute_admin_command(collection, {"getClusterParameter": PARAM_NAME})
+        assertSuccessPartial(
+            result,
+            {"clusterParameters": [{"preAndPostImages": {"expireAfterSeconds": Int64(3600)}}]},
+            msg="Last write should win",
+        )
+    finally:
+        _restore(collection)
+
```
Comment (SaiPradyumn) [inline documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_core_behavior.py:180]: changeStreamOptions has round-trip, idempotent, and last-write-wins coverage, but
pauseMigrationsDuringMultiUpdates only appears in the null-reset test. Consider adding a basic round-trip test for it.

File: documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_errors.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
+"""Tests for setClusterParameter error cases.
```
Comment (SaiPradyumn) [inline documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_errors.py:1]: expireAfterSeconds: -1 is rejected. expireAfterSeconds: 0 is not tested — zero may be accepted (as "expire immediately") or rejected (same as negative). It's a boundary value worth pinning:

Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_errors.py:1]: added

File: documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_core_behavior.py (hunk 1 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_core_behavior.py:180]: fixed

**Review by SaiPradyumn — APPROVED** [review]

**Review by eerxuan — APPROVED** [review]

File: documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_bson_type_validation.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+
+pytestmark = [pytest.mark.admin, pytest.mark.no_parallel]
+
+# NULL is skipped from the rejection matrices below: its behavior is field-dependent and does not
+# cleanly map to success or failure, so it is covered by dedicated tests in the other files.
+BSON_TYPE_SPECS = [
+    BsonTypeTestCase(
+        id="setClusterParameter_value",
+        msg="setClusterParameter value should only accept object type",
+        keyword="setClusterParameter",
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_bson_type_validation.py:29]: keyword is wrong, restore won't find the keyword.

Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/system/administration/commands/setClusterParameter/test_setClusterParameter_bson_type_validation.py:29]: fixed

**Review by eerxuan — APPROVED** [review]
